adapter/utils/src/main/java/org/mobilitydata/gtfsvalidator/utils/GeodeticUtils.java

```diff
@@ -0,0 +1,110 @@
+
+package org.mobilitydata.gtfsvalidator.utils;
+
+import org.locationtech.spatial4j.distance.DistanceCalculator;
+import org.locationtech.spatial4j.distance.DistanceUtils;
+import org.locationtech.spatial4j.shape.Point;
+import org.locationtech.spatial4j.shape.ShapeFactory;
+
+import static org.locationtech.spatial4j.context.SpatialContext.GEO;
+
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.ShapePoint;
+import org.mobilitydata.gtfsvalidator.usecase.distancecalculationutils.DistanceCalculationUtils;
+import org.mobilitydata.gtfsvalidator.usecase.distancecalculationutils.DistanceUnit;
+
+import java.util.Map;
+
+/**
+ * Interface implementation using external library org.locationtech.spatial4j. This is used to compute distance between
+ * two {@code ShapePoint}.
+ */
+public class GeodeticUtils implements DistanceCalculationUtils {
+
+    /**
+     * Return the distance between two points given there lat/lon positions in the specified unit. Note that points of
+     * origin and destination can be swapped.
+     *
+     * @param originLatitude        latitude of the origin point
+     * @param originLongitude       longitude of the origin point
+     * @param destinationLatitude   latitude of the destination point
+     * @param destinationLongitude  longitude of the destination point
+     * @param distanceUnit          unit of the desired result of computation
+     * @return the distance between two points given there lat/lon positions in the specified unit
+     */
+    @Override
+    public double distanceBetweenTwoPoints(final float originLatitude, final float originLongitude,
+                                           final float destinationLatitude, final float destinationLongitude,
+                                                  final DistanceUnit distanceUnit) {
+        final ShapeFactory shapeFactory= getShapeFactory();
+        final DistanceCalculator distanceCalculator = getDistanceCalculator();
+
+        final Point origin = shapeFactory.pointXY(originLongitude, originLatitude);
+        final Point destination = shapeFactory.pointXY(destinationLongitude, destinationLatitude);
+        // implementation uses harvesine formula which determines the great-circle distance between two points on a
+        // sphere given their longitudes and latitudes. Note that the elevation of both points is not taken into
+        // consideration.
+        final double distance = DistanceUtils.DEG_TO_KM*distanceCalculator.distance(origin, destination);
+        if (distanceUnit == DistanceUnit.KILOMETER) {
+                return distance;
+        } else {
+                return distance* KILOMETER_TO_METER_CONVERSION_FACTOR;
+        }
+    }
+
+    /**
+     * private method returning a {@code ShapeFactory}
+     * @return a {@link ShapeFactory}
+     */
+    private static ShapeFactory getShapeFactory() {
+        return GEO.getShapeFactory();
+    }
+
+    /**
+     * private method returning a {@code DistanceCalculator}
```

Comment:
Generally you don't need to include things like "private method" in the Javadoc which are obvious just from looking at the method signature.